packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/frescosupport/FrescoBasedReactTextInlineImageSpan.kt

```diff
@@ -0,0 +1,173 @@
+    private val draweeHolder: DraweeHolder<GenericDraweeHierarchy> =
+        DraweeHolder(GenericDraweeHierarchyBuilder.newInstance(resources).build())
+
+    override val width: Int
+        get() = _width
+
+    override val height: Int
+        get() = _height
+
+    override var drawable: Drawable? = null
+        private set
+
+    /**
+     * The ReactTextView that holds this ImageSpan is responsible for passing these methods on so
+     * that we can do proper lifetime management for Fresco
+     */
+    public override fun onDetachedFromWindow() {
+        draweeHolder.onDetach()
+    }
+
+    public override fun onStartTemporaryDetach() {
+        draweeHolder.onDetach()
+    }
+
+    public override fun onAttachedToWindow() {
+        draweeHolder.onAttach()
+    }
+
+    public override fun onFinishTemporaryDetach() {
+        draweeHolder.onAttach()
+    }
+
+    public override fun getSize(
+        paint: Paint,
+        text: CharSequence,
+        start: Int,
+        end: Int,
+        fm: Paint.FontMetricsInt?
+    ): Int {
+        // NOTE: This getSize code is copied from DynamicDrawableSpan and modified
+        // to not use a Drawable
+
+        fm?.let {
+            it.ascent = -_height
+            it.descent = 0
+
+            it.top = it.ascent
+            it.bottom = 0
+        }
+
+        return _width
+    }
+
+    public override fun setTextView(textView: TextView?) {
+        this.textView = textView
+    }
+
+    public override fun draw(
+        canvas: Canvas,
+        text: CharSequence,
+        start: Int,
+        end: Int,
+        x: Float,
+        top: Int,
+        y: Int,
+        bottom: Int,
+        paint: Paint
+    ) {
+        if (drawable == null) {
+            val imageRequestBuilder = ImageRequestBuilder.newBuilderWithSource(_uri)
+            val imageRequest: ImageRequest =
+                ReactNetworkImageRequest.fromBuilderWithHeaders(imageRequestBuilder, headers)
+
+            draweeHolder.hierarchy.setActualImageScaleType(ImageResizeMode.toScaleType(resizeMode))
+
+            draweeControllerBuilder.reset()
+            draweeControllerBuilder.oldController = draweeHolder.controller
+
+            callerContext?.let { draweeControllerBuilder.setCallerContext(it) }
+
+            draweeControllerBuilder.setImageRequest(imageRequest)
+
+            val draweeController = draweeControllerBuilder.build()
+            draweeHolder.controller = draweeController
+            draweeControllerBuilder.reset()
+
+            checkNotNull(draweeHolder.topLevelDrawable).apply {
+                setBounds(0, 0, _width, _height)
+                if (tintColor != 0) {
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                        colorFilter = BlendModeColorFilter(tintColor, BlendMode.SRC_IN)
+                    } else {
+                        setColorFilter(tintColor, PorterDuff.Mode.SRC_IN)
+                    }
```

Comment:
@mateoguzmana I had added that sdk version check to account for the function deprecation. In my experience in other projects, usually the warning resolves with that check. But still getting:

```
> Task :packages:react-native:ReactAndroid:configureCMakeRelWithDebInfo[arm64-v8a]
e: warnings found and -Werror specified
w: file:///__w/react-native/react-native/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/frescosupport/FrescoBasedReactTextInlineImageSpan.kt:151:25 'fun setColorFilter(p0: Int, p1: PorterDuff.Mode): Unit' is deprecated. Deprecated in Java.
```

Apart from that warning app did compile and run fine locally.

Reply:
You can try:

```suggestion
                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
                        colorFilter = BlendModeColorFilter(tintColor, BlendMode.SRC_IN)
                    } else {
                        colorFilter = PorterDuffColorFilter(tintColor, PorterDuff.Mode.SRC_IN)
                    }
```

In case we need to keep it as-is, you can add on top of the line `@Suppress("DEPRECATION")` for now.

Reply:
I dont get the issue locally now. I think should be fine on CI too 👍